log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java

```diff
@@ -812,6 +812,22 @@ private void initApiModule() {
                 .init(); // Or make public and call ThreadContext.init() which calls ThreadContextMapFactory.init().
     }
 
+    private Logger newInstance(final String name, final MessageFactory messageFactory) {
+        final Logger logger = newInstance(this, name, messageFactory);
+        final MessageFactory loggerMessageFactory = logger.getMessageFactory();
+        if (!loggerMessageFactory.equals(messageFactory)) {
+            StatusLogger.getLogger()
+                    .error(
+                            "Newly created logger with name `{}` and message factory `{}` was actually requested to be created with a different message factory: `{}`.\n"
```

Comment:
Why should these differ?

Reply:
They shouldn't, but API allows this corner case. I wouldn't be surprised to see a `LC#newInstance()` implementation ignoring the passed `MF`.